### ISSUE TITLE
Numpy unsigned int values evaluation

### ISSUE THREAD

**Issue by DavidLP** [numba/numba#1653]
When you use unsigned int to int conversion followed by an absolute calculation the jitted code behaves differently than the pure python code.

Example:

Input:
```python
@njit
def test(hits):
        if abs(int(np.uint16(1)) - int(np.uint16(2))) <= 1:
            print(1)
        else:
            print(0)
```

Output: 0

The pure python version gives as an output 1.

**Comment by pitrou** [comment]
Indeed. You can replace the call to int() to a call to an explicitly sized int such as `np.int32()` to get the explicit behaviour.
I don't know what the behaviour of int() should be in Numba code. Should it convert to a signed it? Should it be a no-op on integers? Should it convert to a known bitsize?

**Comment by seibert** [comment]
Probably the right thing to do is to cast up to the size of normal, non-bigint, Python integers (are they 64-bit everywhere, or just on 64-bit platforms?).

**Comment by pitrou** [comment]
Well, they don't exist anymore on Python 3 :-)
On Python 2 they're a C long, which is a bit more complicated than "32-bit on 32-bit platforms, 64-bit on 64-bit platforms" :-) (they're 32-bit on 64-bit Windows, for example)

**Comment by seibert** [comment]
Right, but we still treat ints in Python 3 as 64-bit since we don't have a bigint implementation.  Worst case, we map everything to int64 and just eat the slight performance loss on 32-bit.

**Comment by seibert** [comment]
Confirmed this still happens in Numba 0.35.
